### PR TITLE
fix(web): replace Bootstrap 3.3.7 with 5.3.3

### PR DESCRIPTION
**Pull request by tnir**
## Background

While Sidekiq Web UI has used Boostrap 3.3.7 since #3117 (2016; 8 years ago), 5 years passed after the EOL of Bootstrap 3.

> Bootstrap 3 reached end of life July 24, 2019, followed by Bootstrap 4 on January 1, 2023. As such, neither version receives new features, updates, or security fixes. 

https://getbootstrap.com/docs/4.6/end-of-life/

## Changes

- Replaces Bootstrap 3.3.7 with 5.3.3.

## Refs.

- #3117
- #4103
- Bootstrap's migration guides:
   - https://getbootstrap.com/docs/4.6/migration/
   - https://getbootstrap.com/docs/5.3/migration/

### CODE REVIEW

**Comment by mperham** [comment]
Testing:

- [ ] RTL languages
- [ ] Dark/light mode
- [ ] Pro/Ent support
- [ ] Live Polling

**Comment by mperham** [comment]
Screenshot 2024-10-07 at 9 40 40 AM

**Comment by mperham** [comment]
RTL looks good enough. Dark mode ❌.

Screenshot 2024-10-07 at 10 21 22 AM

**Comment by mperham** [comment]
Minor issues with Pro/Ent which I can fix, e.g. rate limiters filtering widget.

**Comment by mperham** [comment]
Thank you for doing this work. I attempted this upgrade last Spring and it got overwhelming as I'm not good with CSS/styling so this is a huge lift for me.

File: web/views/_poll_link.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <% if pollable? %>
     <a class="live-poll-start live-poll btn btn-primary"><%= t('LivePoll') %></a>
-    <a class="live-poll-stop live-poll btn btn-primary active"><%= t('StopPolling') %></a>
+    <a class="live-poll-stop live-poll btn btn-primary active" style="display: none"><%= t('StopPolling') %></a>
```
Comment (markets) [inline web/views/_poll_link.erb:3]: @tnir @mperham I think we should remove inline styles: #6292

**Comment by markets** [comment]
@mperham I understand this change, as Sidekiq needs to move forward also regarding its front-end dependencies, but my question is: How this can be handled by plugins with UI? I'm the maintainer of [`sidekiq-cron`](https://github.com/sidekiq-cron/sidekiq-cron) (one of the most used Sidekiq plugins) and for the views we are just reusing current Bootstrap classes, example: https://github.com/sidekiq-cron/sidekiq-cron/blob/master/lib/sidekiq/cron/views/cron.erb. So after this Bootstrap update, it will really hard to maintain compatibility with different Sidekiq versions.

**Comment by mperham** [comment]
@markets I appreciate your compatibility concerns. We'll get this into Sidekiq 8.0 so it is part of a major version change. If you want to maintain compatibility with both 7 and 8 in the same package, that's up to you. You only have two pages so you could provide two templates, one for 7 and one for 8, and render based on Sidekiq::MAJOR.

**Comment by markets** [comment]
Yes, the only reasonable way I can imagine right now is to provide two templates. Out of curiosity, do you have any ETA for the v8 release? Depending on the changes it will be probably better to jump a major version on our side too. We already implemented support for the new UI assets API (https://github.com/sidekiq-cron/sidekiq-cron/pull/480), but that was easy to make it work with different Sidekiq versions.

Thanks Mike!

**Comment by mperham** [comment]
> do you have any ETA for the v8 release?

Not in 2024. 2025ish. 😂

**Comment by mperham** [comment]
There's still many noticeable regressions in the look and feel from stock. Here's stock:

stock

and here's this PR:

pr

I hate to nitpick contributions but can you go through the pages Stock vs PR and clean up as many differences as possible? Part of the reason why Sidekiq is so popular is that we care about this level of detail.

**Comment by mperham** [comment]
The two major issues in those screenshots are the upper navbar tab highlighting and the smaller action buttons in the Processes table. Hopefully easy fixes.

**Comment by mperham** [comment]
See #6549 and #6551

**Pull request closed without merging**